dataflowengineoss/src/main/scala/io/joern/dataflowengineoss/queryengine/TaskCreator.scala

```diff
@@ -93,49 +101,71 @@ class TaskCreator(context: EngineContext) {
-      val outCall = outArg.collect { case n: Call => n }
-
-      val methodReturns = outCall.toList
-        .flatMap(x => NoResolve.getCalledMethods(x).methodReturn.map(y => (x, y)))
-        .iterator
-
-      methodReturns.flatMap { case (call, methodReturn) =>
-        val method           = methodReturn.method
-        val returnStatements = methodReturn._reachingDefIn.toList.collect { case r: Return => r }
-        if (method.isExternal || method.start.isStub.nonEmpty) {
-          val newPath = path
-          (call.receiver.l ++ call.argument.l).map { arg =>
-            val taskStack = result.taskStack :+ TaskFingerprint(arg, result.callSiteStack, callDepth)
-            ReachableByTask(taskStack, newPath)
-          }
-        } else {
-          returnStatements.map { returnStatement =>
-            val newPath = Vector(PathElement(methodReturn, result.callSiteStack)) ++ path
-            val taskStack =
-              result.taskStack :+ TaskFingerprint(returnStatement, call :: result.callSiteStack, callDepth + 1)
-            ReachableByTask(taskStack, newPath)
+      outArg.toList.flatMap {
+        case call: Call =>
+          val methodReturns = call.toList
+            .flatMap(x => NoResolve.getCalledMethods(x).methodReturn.map(y => (x, y)))
+            .iterator
+
+          methodReturns.flatMap { case (call, methodReturn) =>
+            val method           = methodReturn.method
+            val returnStatements = methodReturn._reachingDefIn.toList.collect { case r: Return => r }
+            if (method.isExternal || method.start.isStub.nonEmpty) {
+              val newPath = path
+              (call.receiver.l ++ call.argument.l).map { arg =>
+                val taskStack = result.taskStack :+ TaskFingerprint(arg, result.callSiteStack, callDepth)
+                ReachableByTask(taskStack, newPath)
+              }
+            } else {
+              returnStatements.map { returnStatement =>
+                val newPath = Vector(PathElement(methodReturn, result.callSiteStack)) ++ path
+                val taskStack =
+                  result.taskStack :+ TaskFingerprint(returnStatement, call :: result.callSiteStack, callDepth + 1)
+                ReachableByTask(taskStack, newPath)
+              }
+            }
           }
-        }
+        case _ => Vector.empty
       }
     }
 
     val forArgs = outArgsAndCalls.flatMap { case (result, args, path, callDepth) =>
-      args.toList.flatMap { case arg: Expression =>
-        val outParams = if (result.callSiteStack.nonEmpty) {
-          List[MethodParameterOut]()
-        } else {
-          argToOutputParams(arg).l
-        }
-        outParams
-          .filterNot(_.method.isExternal)
-          .map { p =>
-            val newStack = arg.inCall.headOption.map { x => x :: result.callSiteStack }.getOrElse(result.callSiteStack)
-            ReachableByTask(result.taskStack :+ TaskFingerprint(p, newStack, callDepth + 1), path)
+      args.toList.flatMap {
```

Comment:
From here to line 150, are there any changes or is this just formatting?

Reply:
Yes, only formatting and handling of `case _ => Vector.empty` for nonExpression nodes